pkg/terraform/terraform.go

```diff
@@ -111,13 +113,143 @@ type TerraformEntrypoint struct {
 	BackendFile string
 }
 
+type ModuleUsageGraph struct {
+	EntrypointToModules  map[string]map[string]struct{}
+	ModulesToEntrypoints map[string]map[string]struct{}
+}
+
+// moduleSourcePattern is a Regex pattern used to a module source from the module block in a terraform config.
+var moduleSourcePattern = regexp.MustCompile(`source\s*\=\s\"(.*)\"`)
```

Comment:
Can strings be single-quoted in HCL? I seem to recall that they can, but I could be mistaken.

Reply:
Resolved by switching to hcl parser.